authservice/auth_service.go

```diff
@@ -196,6 +208,16 @@ func NewAuthService(serviceID string, serviceHost string, serviceLoader ServiceR
 	return auth, nil
 }
 
+func (a *AuthService) CheckForRefresh() (bool, error) {
+	var loadServicesError error
+	now := time.Now()
+	if a.servicesUpdated == nil || now.Sub(*a.servicesUpdated).Minutes() > float64(a.minRefreshCacheFreq) {
```

Comment:
Sorry my previous comment on the locks here was misleading. We do still need to acquire the read lock before we can access the `serviceUpdated` property. I think what we should be doing here is locking before the line 214 and unlocking before line 215 and before line 218. 

Alternatively it may make this much simpler to lock, store `serviceUpdated` and `minRefreshCacheFreq` to local variables at the beginning of the function, then unlock and check the local values instead on line 214. We could use a similar strategy above as well.